openprocurement/tender/core/tests/tender.py

```
@@ -67,3 +75,322 @@ def test_empty_listing(self):
+                {u'description': u'Not implemented', u'location': u'data', u'name': u'procurementMethodType'}
+            ])
+
+        for _ in range(5):
+            response = self.app.get('/tenders')
+            self.assertEqual(response.status, '200 OK')
+            self.assertEqual(len(response.json['data']), 0)
+
+
+class BelowThresholdTenderResourceTest(BaseWebTest):
+    """ Test case that adds 'belowThreshold' plugin
+    to test tender creating.
+    """
+    initial_auth = ('Basic', ('broker', ''))
+    relative_to = os.path.dirname(__file__)
+
+    @classmethod
+    def setUpClass(self):
+        # overriding parent's method to add belowThreshold for test case
+        self.app = TestApp(
+            get_app('{}/tests.ini'.format(self.relative_to), options={
+                'plugins': 'api,tender_core,belowThreshold'}),
+            relative_to=self.relative_to
+        )
+        self.app.RequestClass = PrefixedRequestClass
+        self.couchdb_server = self.app.app.registry.couchdb_server
+        self.db = self.app.app.registry.db
+        self.db_name = self.db.name
+
+    def test_listing(self):
+        response = self.app.get('/tenders')
+        self.assertEqual(response.status, '200 OK')
+        self.assertEqual(len(response.json['data']), 0)
+
+        tenders = []
+
+        for _ in range(3):
+            offset = get_now().isoformat()
+            response = self.app.post_json('/tenders', {'data': test_tender_data})
+            self.assertEqual(response.status, '201 Created')
+            self.assertEqual(response.content_type, 'application/json')
+            tenders.append(response.json['data'])
+
+        ids = set([i['id'] for i in tenders])
+        self.assertEqual(len(ids), 3)
+
+        for _ in range(5):
+            try:
+                response = self.app.get('/tenders')
+                self.assertEqual(len(response.json['data']), 3)
+            except AssertionError:
```

except pass не дуже правильна практика